### ISSUE TITLE
NamedModulesPlugin produces colliding module ids

### ISSUE THREAD

**Issue by jhnns** [webpack/webpack#7088]
**Do you want to request a *feature* or report a *bug*?**

Bug, probably the cause for https://github.com/boopathi/react-svg-loader/issues/197

**What is the current behavior?**

Webpack links modules differently in development mode then in production mode depending on the loader configuration.

If I have two importing modules:

```js
// checkA.js
import A from "./other.js";
```

```js
// checkB.js
import B from "./other.js";
```

and apply two different loader configurations on `other.js` like

```
    module: {
        rules: [
            {
                include: require.resolve("./src/other.js"),
                issuer: require.resolve("./src/checkA.js"),
                use: require.resolve("./loaders/a.js")
            },
            {
                include: require.resolve("./src/other.js"),
                issuer: require.resolve("./src/checkB.js"),
                use: require.resolve("./loaders/b.js")
            }
        ]
    },
```

`checkB.js` receives the loader result from `loaders/a.js`

**If the current behavior is a bug, please provide the steps to reproduce.**

https://github.com/jhnns/webpack-multi-loader-bug

**What is the expected behavior?**

`checkA` and `checkB` should receive different results depending on their loader configuration

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

The problem probably exists in older webpack versions. But with webpack 4, it has becoming more urgent since the `NamedModulesPlugin` is applied by default in development mode.

**Comment by jhnns** [comment]
If I change the call in [`libIdent` in `NormalModule`](https://github.com/webpack/webpack/blob/941be2941e5fa51136454158ea812000c78c4a08/lib/NormalModule.js#L114) from:

```js
libIdent(options) {
	return contextify(options.context, this.userRequest);
}

```

to

```js
libIdent(options) {
	return contextify(options.context, this.request);
}
``` 

it works. But that has other consequences I guess... maybe you could clarify what `libIdent` is supposed to be? @sokra

**Comment by sokra** [comment]
Yep that's an issue. libIdent is correct. The plugin need to handle conflicts.

**Comment by ooflorent** [comment]
Looks like this was fixed in 4.7.0 by #7152. Do you still encounter the issue?
